db/migrate/20190405160400_remove_duplicate_tags.rb

```diff
@@ -0,0 +1,8 @@
+class RemoveDuplicateTags < ActiveRecord::Migration[5.1]
+  def change
+    duplicate_row_values = Tagging.select('project_id, tag_id, count(*)').group('project_id, tag_id').having('count(*) > 1').pluck(:project_id, :tag_id)
+    duplicate_row_values.each do |project_id, tag_id|
+      Tagging.where(project_id: project_id, tag_id: tag_id).order(id: :desc)[1..-1].map(&:destroy)
+    end
+  end
```

Comment:
@vik-y @tachyons can you review if this migration is correct?

Reply:
Difficult to comment without testing it

Reply:
@tachyons though there may be no way for you to know for sure whether it will delete the duplicates but from the code alone you can see that it surely isn't going to touch any other critical data (because we are only select and destroying taggings ). Also, if you have time you can run it on a test database (I have and it works fine).